Modules/StudyProgramme/classes/model/AutoMemberships/class.ilStudyProgrammeMembershipSourceReaderOrgu.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types = 1);
+
+/**
+ * Provides adapters to read member-ids from a specific source.
+ */
+class ilStudyProgrammeMembershipSourceReaderOrgu implements ilStudyProgrammeMembershipSourceReader
+{
+	public function __construct(
+		ilObjOrgUnitTree $orgu_tree,
+		ilOrgUnitUserAssignment $orgu_assignment,
+		int $src_id
```

Comment:
Gibt es einen speziellen grund, warum src-id hier schon definiert werden muss? Warum nicht als Parameter in getMemberIds? Dann könnte man das objekt auch vom dic handeln lassen.

Reply:
Das ist richtig.
Ich wollte eine Gleichförmigkeit der SourceReaders, und SourceReaderParticipants  wird mit einer Instanz von ilParticipants gebaut, die bereits die src_id benötigt.  Ich wollte alle Dependencies in der Factory lassen...
Vielleicht ist das ganze Reader-Zeug etwas zuviel, aber es lohnt sich, glaube ich,  bei Ergänzungen um weitere User-Quellen.